packages/legend-application-query/src/stores/QueryBuilderExplorerState.ts

```diff
@@ -370,6 +372,14 @@ export const getQueryBuilderPropertyNodeData = (
     isPartOfDerivedPropertyBranch,
     mappingNodeData,
   );
+  if (
```

Comment:
I think we should do the analytics differently:
1. We know for association case the loop is of size 2, so we should add that analytics together with the fact that the property comes from `propertiesFromAssociations`, so we would have something like this

```


> Firm
    > employees: Person
        // firm: Firm --> we should not show this
        name
        age

```
2. We should not even generate the node, instead of using `mapped` flag to show/hide it, that's not the purpose of the flag